### PR TITLE
feat: POC 1 — Context Injection & Plan Quality

### PR DESCRIPTION
**Pull request by leehopper**
## Summary

- **Deterministic Training Science Layer**: VDOT calculator (Daniels/Gilbert formula), pace calculator (Daniels' tables with interpolation), 18 data model types, 5 complete test profiles with simulated training history
- **Coaching Prompt & Context Assembly**: System prompt YAML with coaching persona/safety rules/output format, context injection template with positional layout (U-curve optimized), ContextAssembler with ~15K token budget enforcement, ICoachingLlm adapter via Anthropic .NET SDK, console CLI app for prompt iteration
- **Eval Suite with Safety Assertions**: 5 plan generation eval tests (beginner volume limits, VDOT-derived pace validation, maintenance volume, injury constraints, scheduling constraints) + 5 safety boundary eval tests (medical, overtraining, injury, crisis, nutrition) — currently skipped pending dedicated LLM testing workflow
- **Context Injection Experiments**: Infrastructure for 4 experiments (token budget, positional placement, summarization level, conversation history), experiment runner with dry-run mode, findings document template, coaching-v2.yaml demonstrating prompt iteration

**Stats**: 137 files, ~14K lines added, 285 tests passing, 11 skipped (live API), 0 failures

## Test plan

- [x] `dotnet build` — 0 warnings, 0 errors across 3 projects
- [x] `dotnet test` — 285 passed, 11 skipped, 0 failed
- [x] Model ID updated to `claude-sonnet-4-5-20250514`
- [x] No secrets in any committed file
- [x] Validation report passes all 6 gates (30/30 requirements, 44/44 proof artifacts)
- [ ] Live API eval tests to be enabled in a follow-up LLM testing workflow
- [ ] Experiments to be run against live API with findings populated

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: backend/src/RunCoach.Api/Modules/Coaching/ContextAssembler.cs [hunk 1]
```diff
@@ -0,0 +1,607 @@
...
+    /// 4. Reduce training history to most recent 2 weeks only
+    /// 5. Truncate conversation to most recent 3 turns.
+    /// </summary>
+    private (List<PromptSection> Middle, List<PromptSection> End, int TotalTokens) ApplyOverflowCascade(
+        ContextAssemblerInput input,
+        List<PromptSection> startSections,
+        List<PromptSection> middleSections,
+        List<PromptSection> endSections,
+        string systemPrompt)
+    {
```
Comment (leehopper) [inline backend/src/RunCoach.Api/Modules/Coaching/ContextAssembler.cs:310]: 🔴 **[critical] [test-coverage]** Overflow cascade has zero test coverage

The 5-step ApplyOverflowCascade (lines 304-374) is never exercised by any test. All profiles fit under 15K tokens. This is the core budget enforcement mechanism.

**Suggested fix:** Add tests with 50+ workouts and 20+ turns to exceed budget, then assert each cascade step.

---
*Confidence: 95% | deep-review*

File: backend/src/RunCoach.Api/Modules/Coaching/ClaudeCoachingLlm.cs [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+        var stopReason = response.StopReason?.ToString() ?? "unknown";
+        LogReceivedResponse(
+            _logger,
+            responseModel,
+            json.Length,
+            stopReason,
+            response.Usage.InputTokens,
+            response.Usage.OutputTokens);
+
+        return JsonSerializer.Deserialize<T>(json, StructuredOutputSerializerOptions)
```
Comment (leehopper) [inline backend/src/RunCoach.Api/Modules/Coaching/ClaudeCoachingLlm.cs:184]: 🔴 **[critical] [test-coverage]** GenerateStructuredAsync deserialization failure untested

No test covers: (1) JSON null literal, (2) malformed JSON, (3) type mismatch. Only happy path tested.

**Suggested fix:** Add tests for null literal, malformed JSON, and empty response.

---
*Confidence: 92% | deep-review*

File: backend/src/RunCoach.Api/Modules/Coaching/ClaudeCoachingLlm.cs [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+            ],
+            Temperature = _settings.Temperature,
+        };
+
+        var response = await _client.Messages.Create(createParams, ct).ConfigureAwait(false);
+
+        var text = ExtractTextContent(response);
+
+        var responseModel = response.Model.ToString();
+        var stopReason = response.StopReason?.ToString() ?? "unknown";
```
Comment (leehopper) [inline backend/src/RunCoach.Api/Modules/Coaching/ClaudeCoachingLlm.cs:120]: 🟠 **[high] [bug+conventions]** stop_reason logged but never checked for truncation

stop_reason is captured and logged but never compared against `max_tokens`. Truncated responses silently returned.

**REVIEW.md rule:** *Always check stop_reason on LLM API responses.*

**Suggested fix:** After logging, throw if `response.StopReason` is `max_tokens`.

---
*Confidence: 100% (consensus) | deep-review*

File: backend/src/RunCoach.Api/Modules/Coaching/Prompts/PromptStoreSettings.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    /// Gets the base directory path for YAML prompt files.
+    /// Defaults to "Prompts" (relative to the application content root).
+    /// </summary>
+    public string BasePath { get; init; } = "Prompts";
+
+    /// <summary>
+    /// Gets the mapping of prompt IDs to their active version strings.
+    /// Example: { "coaching-system": "v1" }.
+    /// </summary>
+    public Dictionary<string, string> ActiveVersions { get; init; } = new();
```
Comment (leehopper) [inline backend/src/RunCoach.Api/Modules/Coaching/Prompts/PromptStoreSettings.cs:24]: 🟠 **[high] [type-design]** Mutable Dictionary on sealed record

`IOptions<T>` injects the same instance across the pipeline. Mutation by one consumer affects all others.

**Suggested fix:** Change to `IReadOnlyDictionary<string, string>`.

---
*Confidence: 88% | deep-review*

**Review by leehopper — COMMENTED** [review]
### Deep Review

**Verdict:** REQUEST CHANGES

Found **21 issues** (2 critical, 6 high, 10 medium, 3 low). See inline comments for top 8 findings.

---
Generated by deep-review | Reviewed up to: 9e3c6bb911c74a2b18a18f1b467ccb62da97e102

File: backend/src/RunCoach.Api/Modules/Coaching/ContextAssembler.cs [hunk 5]
```diff
@@ -0,0 +1,607 @@
...
+        List<PromptSection> end,
+        string systemPrompt)
+    {
+        return EstimateTokens(systemPrompt) + SumTokens(start) + SumTokens(middle) + SumTokens(end);
+    }
+
+    private PromptSection BuildUserProfileSection(UserProfile profile)
+    {
+        var sb = new StringBuilder();
+        sb.AppendLine(CultureInfo.InvariantCulture, $"Name: {profile.Name}");
```
Comment (leehopper) [inline backend/src/RunCoach.Api/Modules/Coaching/ContextAssembler.cs:388]: 🟡 **[medium] [security]** Prompt injection via unsanitized user fields

User-controlled strings (Name, Constraints[], InjuryNote.Description, Notes, Messages) interpolated directly into LLM prompts without sanitization or length limits.

**Suggested fix:** Wrap user content in XML tags with system prompt instructions to treat as data. Add length limits at API boundary.

---
*Confidence: 78% | deep-review*

File: backend/src/RunCoach.Api/Modules/Coaching/Prompts/YamlPromptStore.cs [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+            return new Lazy<Task<PromptTemplate>>(
+                () => LoadAsync(id, version, CancellationToken.None));
+        });
+
+        if (!isNewEntry)
+        {
+            LogCacheHit(_logger, id, version);
+        }
+
+        try
```
Comment (leehopper) [inline backend/src/RunCoach.Api/Modules/Coaching/Prompts/YamlPromptStore.cs:95]: 🟠 **[high] [test-coverage]** Faulted-cache-eviction path untested

The catch clause that evicts faulted entries via `TryRemove` has no test. A transient I/O failure could permanently poison the cache.

**Suggested fix:** Test: call with missing file, catch, create file, call again, assert success.

---
*Confidence: 88% | deep-review*

File: backend/src/RunCoach.Api/Modules/Coaching/Prompts/YamlPromptStore.cs [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+    [LoggerMessage(Level = LogLevel.Information, Message = "Loading prompt file for '{Id}' version '{Version}' from '{FilePath}'")]
+    private static partial void LogLoadingFile(ILogger logger, string id, string version, string filePath);
+
+    [LoggerMessage(Level = LogLevel.Information, Message = "Loaded and cached prompt template '{Id}' version '{Version}'")]
+    private static partial void LogLoadedTemplate(ILogger logger, string id, string version);
+
+    [LoggerMessage(Level = LogLevel.Information, Message = "Validated prompt '{Id}' version '{Version}' exists at '{FilePath}'")]
+    private static partial void LogValidatedVersion(ILogger logger, string id, string version, string filePath);
+
+    private string BuildFilePath(string id, string version) =>
```
Comment (leehopper) [inline backend/src/RunCoach.Api/Modules/Coaching/Prompts/YamlPromptStore.cs:179]: 🟡 **[medium] [security]** Path traversal possible in BuildFilePath

id and version validated against `::` but not path traversal (`../`). Currently safe but `IPromptStore` interface accepts arbitrary strings.

**Suggested fix:** Verify `Path.GetFullPath(filePath).StartsWith(_basePath)`.

---
*Confidence: 75% | deep-review*

File: backend/src/RunCoach.Api/Modules/Training/Models/DecimalRange.cs [hunk 8]
```diff
@@ -0,0 +1,7 @@
+namespace RunCoach.Api.Modules.Training.Models;
+
+/// <summary>
+/// A generic numeric range with minimum and maximum decimal values.
+/// Used for distance ranges, volume targets, and similar bounded quantities.
+/// </summary>
+public sealed record DecimalRange(decimal Min, decimal Max);
```
Comment (leehopper) [inline backend/src/RunCoach.Api/Modules/Training/Models/DecimalRange.cs:7]: 🟡 **[medium] [type-design]** DecimalRange allows inverted ranges (Min > Max)

`new DecimalRange(100, 10)` compiles but is semantically invalid. Same issue in `PaceRange`.

**Suggested fix:** Add `ArgumentOutOfRangeException.ThrowIfGreaterThan(Min, Max)` in constructor.

---
*Confidence: 82% | deep-review*
